### PR TITLE
Event driven joypad

### PR DESCRIPTION
**Pull request by aerydna**
The Joypad EventDriven mode can be activated when using the `IJoypadController` interface by providing a valid pointer to a `IJoypadEvent` object. A thread will start and will call the `IJoypadEvent::action` method containing the detected events at every `RateThread::run`.

### CODE REVIEW

File: src/libYARP_dev/include/yarp/dev/IJoypadController.h [hunk 1]
```diff
@@ -155,6 +171,97 @@ class  YARP_dev_API yarp::dev::IJoypadController
...
+    virtual bool getRawAxis(unsigned int axis_id, double& value) = 0;
+    virtual bool getRawStick(unsigned int stick_id, yarp::sig::Vector& value, JoypadCtrl_coordinateMode coordinate_mode) = 0;
+    virtual bool getRawTouch(unsigned int touch_id, yarp::sig::Vector& value) = 0;
+    using IJoypadController::m_actions;
+    using IJoypadController::executeAction;
+    using IJoypadController::parseActions;
+
+public:
+
+    virtual bool getAxisCount(unsigned int& axis_count) YARP_OVERRIDE YARP_FINAL;
```
Comment (drdanz) [inline src/libYARP_dev/include/yarp/dev/IJoypadController.h]: I'm not 100% sure, but I think that `override final` is redundant, `final` implies `override`

Comment (aerydna) [inline src/libYARP_dev/include/yarp/dev/IJoypadController.h]: I thought so too.. but it seems not. compiler gives you the Wsuggest-override warning (with only final) and you can put final in a virtual method also in the first declaration (the one in the top class).. it doesn't make any sense to me but is possible..
so you have to specify final and override..

Comment (aerydna) [inline src/libYARP_dev/include/yarp/dev/IJoypadController.h]: https://stackoverflow.com/questions/29412412/does-final-imply-override

**Comment by drdanz** [comment]
Merged thanks.

**Pull request merged**